File: fapi/src/main/java/com/netcracker/edu/project/fapi/model/AuthUser.java
package com.netcracker.edu.project.fapi.model;


public class AuthUser {
    private int id;
    private String name;
    private String surname;
    private String email;
    private RoleModel role;

    public AuthUser() {
    }

    public AuthUser(int id, String name, String surname, String email, RoleModel role) {
        this.id = id;
        this.name = name;
        this.surname = surname;
        this.email = email;
        this.role = role;
    }

    public String getSurname() {
        return surname;
    }

    public void setSurname(String surname) {
        this.surname = surname;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() { return name;}

    public void setName(String name) { this.name = name;}

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public RoleModel getRole() {
        return role;
    }

    public void setRole(RoleModel role) {
        this.role = role;
    }

}

File: frontend/src/app/model/authuser.ts
import {RoleModel} from "./rolemodel";

export class AuthUserModel {
  id: string;
  name: string;
  surname: string;
  email: string;
  role: RoleModel;

  static cloneBase(user: AuthUserModel): AuthUserModel {
    const cloneUserModel: AuthUserModel = new AuthUserModel();
    cloneUserModel.id = user.id;
    cloneUserModel.email = user.email;
    cloneUserModel.name = user.name;
    cloneUserModel.surname = user.surname;
    cloneUserModel.role = user.role;
    return cloneUserModel;
  }


}

export class UserWithPassword{
  password:string;
  name:string;
  surname:string;
  role:string;
  email:string;
}
